src/NuGet.Core/NuGet.Protocol/Resources/OwnerDetailsUriTemplateResourceV3.cs

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using NuGet.Protocol.Core.Types;
+
+namespace NuGet.Protocol.Resources
+{
+    /// <summary>Owner Details Uri Template for NuGet V3 HTTP feeds.</summary>
+    /// <remarks>Not intended to be created directly. Use <see cref="SourceRepository.GetResourceAsync{T}(CancellationToken)"/>
+    /// with <see cref="OwnerDetailsUriTemplateResourceV3"/> for T, and typecast to this class.
+    public class OwnerDetailsUriTemplateResourceV3 : INuGetResource
+    {
+        private readonly string _template;
+
+        private OwnerDetailsUriTemplateResourceV3(string template)
+        {
+            _template = template ?? throw new ArgumentNullException(nameof(template));
+        }
+
+        /// <summary>
+        /// Creates the specified Owner Details Uri template provided by the server if it exists and is valid.
+        /// </summary>
+        /// <param name="uriTemplate">The Absolute Uri template provided by the server.</param>
+        /// <returns>A valid Owner Details Uri template, or null.</returns>
+        public static OwnerDetailsUriTemplateResourceV3? CreateOrNull(Uri? uriTemplate)
+        {
+            if (uriTemplate == null || uriTemplate.OriginalString.Length == 0 || !uriTemplate.IsAbsoluteUri)
+            {
+                return null;
+            }
+
+            string absoluteUri = uriTemplate.OriginalString;
+            if (string.IsNullOrWhiteSpace(absoluteUri)
+                || !IsValidUriTemplate(absoluteUri))
+            {
+                return null;
+            }
+
+            return new OwnerDetailsUriTemplateResourceV3(absoluteUri);
+        }
+
+        private static bool IsValidUriTemplate(string absoluteUri)
```

Comment:
You have a Uri type in the method calling this, so you shouldn't need to pass a string. 

I'd just all the validations in 1 method and use the Uri type that's already there.

Reply:
Hm, I was trying not to invent any new validation strategy, as mentioned in https://github.com/NuGet/NuGet.Client/pull/5763#discussion_r1581410514, [PackageDetailsUriResourceV3](https://github.com/NuGet/NuGet.Client/blob/784274cc9b2002ca73055e8134064261118d8a01/src/NuGet.Core/NuGet.Protocol/Resources/PackageDetailsUriResourceV3.cs#L19) does this custom validation, and it's why I copied that for the `OwnerDetailsUriTemplateResourceV3` as well. 

I'll see if I can reduce this validation.

Reply:
Ok, I don't like this any better than what was there, but now I'm throwing if a null is provided, and now must check for null in the Provider. We don't want providers to throw when they can't find the resource, because it's not required for a package source to implement this resource.